darwin/dataset/upload_manager.py

```diff
@@ -106,6 +108,20 @@ def _chunk_filenames(files: List[Path], size: int):
     No batch has any duplicates with regards to file name.
     This is needed due to a limitation in the upload api.
 
+    This code is a bit weird but this is the general gist of it:
```

Comment:
I would personally keep documentation a little more formal, while keeping more explanatory and informal language in comments, when necessary

Reply:
Addressed